objects/src/assets/token_symbol.rs

```diff
@@ -0,0 +1,79 @@
+use super::{AssetError, Felt, StarkField, ToString};
+use crate::utils::string::String;
+
+const MAX_SUMBOL_LENGHT: usize = 6;
+#[derive(Clone, Copy, Debug)]
+pub struct TokenSymbol(Felt);
+
+impl TokenSymbol {
+    pub fn new(symbol: &str) -> Result<Self, AssetError> {
+        let felt = encode_symbol_to_felt(symbol)?;
+        Ok(Self(felt))
+    }
+
+    pub fn as_felt(&self) -> Felt {
+        self.0
+    }
+
+    pub fn as_symbol(&self) -> Result<String, AssetError> {
+        decode_felt_to_symbol(self.0)
+    }
```

Comment:
I'd change this to `to_symbol()` and make the return type just `String`. This assumes that we always construct `TokenSymbol` with a valid internal value - but I think that's currently the case.